### PR TITLE
`CanonicalizedMap`: new `copy`, `toMap` and `toMapOfCanonicalKeys` methods

### PR DESCRIPTION
**Pull request by gmpassos**
- `copy`: copies a `CanonicalizedMap` instance without recalculating the canonical values of the keys.
- `toMap`: creates a `Map<K,V>` (with the original key values).
- `toMapOfCanonicalKeys`: creates a `Map<C,V>` (with the canocalized keys).

### CODE REVIEW

**Comment by natebosch** [comment]
These APIs look relatively straightforward and I don't have concerns about adding them.

For clarity, @gmpassos - could you describe the use case you have in mind for these?

cc @lrhn

**Comment by gmpassos** [comment]
> It should just be a minor version increment, and a changelog entry for that version saying which methods are added.

Done 👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## 1.17.3
```
Comment (natebosch) [inline CHANGELOG.md:1]:
```suggestion
## 1.18.0-wip
```

File: pubspec.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: collection
-version: 1.17.2
+version: 1.17.3
```
Comment (natebosch) [inline pubspec.yaml:2]:
```suggestion
version: 1.18.0-wip
```

**Review by natebosch — APPROVED** [review]

File: pubspec.yaml (hunk 2 above)
Comment (gmpassos) [reply pubspec.yaml:2]: I can commit the suggestion, but please confirm if you really want to jump to `1.18.0` for this merge.

Best regards.

Comment (natebosch) [reply pubspec.yaml]: Yes, for new APIs we rev the second number in the semver version.

**Review by natebosch — APPROVED** [review]

**Pull request merged**